Initial implementation for portfolio website

anna-giang opened the pull request:
Portfolio website with the following pages:

- **index.html page** - "home page" with links to other pages in the portfolio website
- **aboutMe.html page** - contains a button that links to a addRandomFact function in script.js that adds random facts about me to the page
- **projects.html** 
- **blog.html** - each blog post has a button linked to a toggleBlogPost function in script.js that hides/shows the associated blog post on click event
- **gallery.html** 

Layout of pages largely styled using CSS flexboxes.

anna-giang committed:
Update the README.md file to describe the repo.

anna-giang committed:
Add empty pages for portfolio sections

anna-giang committed:
Personalise index.html - add images that link to individual portfolio sections on index.html and section descriptions.

anna-giang committed:
Add text content and random fact generator for aboutMe page

anna-giang committed:
Add content for projects.html page. Showcasing some of my side projects.

anna-giang committed:
Add content for the blog.html page

Initally hide blog posts on render.
Add button to toggle show/hide of blog posts, linked to toggleBlogPost function in script.js.

anna-giang committed:
Add content for gallery.html

10 images arranged in a picture gallery created using CSS flexbox.

anna-giang committed:
Change arrangement of page elements on index, projects and blog

Use flexboxes to arrange each section of the page into rows and columns.

anna-giang committed:
Arrange elements on the aboutMe.html page

anna-giang committed:
Modify second blog post content in blog.html

Second blog post previously contained test content.

anna-giang committed:
Blur linked images on hover and open external links in new tab

Images that can be clicked are "blurred" (reduce opacity) on mouse-over/hover to indicate that they can be clicked.
Also rename "section-pic" to "section-img" for consistency.

anna-giang committed:
Adjust font-family and font-size for all text

anna-giang committed:
Add footer with links to GitHub, LinkedIn and Email

anna-giang committed:
Add padding around buttons to create space from adjacent elements

anna-giang committed:
Style buttons

anna-giang committed:
Fix format of CSS file according to style guide

anna-giang committed:
Fix style.css formatting according to the style guide.

anna-giang committed:
Fix formatting of HTML code according to style guide

anna-giang committed:
Add alternative contents for all images

anna-giang committed:
Fix formatting and comments in script.js and delete unused code.

BipBopBoop commented on portfolio/src/main/webapp/aboutMe.html:
```diff
@@ -0,0 +1,79 @@
+  </head>
+  <body>
+    <div id="content" class="flex-container">
+        <div class="header-div">
```
nit: This looks like a tab, indents should be 2-spaces, please make sure all your indents are 2-spaced.
https://google.github.io/styleguide/htmlcssguide.html#Indentation

anna-giang replied:
Fixed! Missed that section of the style guide 😅

BipBopBoop replied:
No worries, it can be pretty easy to miss things.

Thanks for changing it up :)

BipBopBoop commented on portfolio/src/main/webapp/aboutMe.html:
```diff
@@ -0,0 +1,79 @@
+            <h2 class="subheading-text">Interests and hobbies</h2>
+            <ul class="body-text">
+                <li>Coding competitions</li>
+                <span class="footnote-text">Complete beginner at this.</span>
```
nit: spans shouldn't be nested within `<ul>` directly: https://html.spec.whatwg.org/multipage/grouping-content.html#the-ul-element (only `<li>` elements should be in `<ul>`). You could try something like
```
<li> Coding Competitions
<br><span> Complete beginner at this.</span>
</li>
```

You can use a validator to pick up on things like this: https://validator.w3.org/ (honestly I would probably miss this myself without it).

anna-giang replied:
Fixed! Thanks for your suggestion!

BipBopBoop replied:
Thanks for changing this

BipBopBoop commented on portfolio/src/main/webapp/gallery.html:
```diff
@@ -0,0 +1,46 @@
+    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+  </head>
+  <body>
+    <div id="content">
```
This div needs to be closed (`</div>`)

anna-giang replied:
Fixed!

BipBopBoop replied:
Thanks!

BipBopBoop commented on portfolio/src/main/webapp/aboutMe.html:
```diff
@@ -0,0 +1,79 @@
+        </div>
+    </div>
+  </body>
+</html>
```
nit: Add a new line at the end of all of your files

anna-giang replied:
Done! 

What is the reason why we need to leave the blank line?

BipBopBoop replied:
Thanks!

This is mainly due to UNIX standards for files and because some programs won't be able to interpret files correctly without a new line at the end.
https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

BipBopBoop commented on portfolio/src/main/webapp/index.html:
```diff
@@ -4,15 +4,51 @@
+            <div class="flex-item">
+                <h2 class="subheading-text">Blog</h2>
+                <a class="linked-img" href="blog.html"><img class="section-img" src="/images/covers/blog-cover.jpg" alt="Palace in South Korea"></a>
+                <p class="body-text">Hear about my the more interesting episodes of my otherwise-pretty-average daily life.</p>
```
nit: typo "Hear about *my* the more..."

anna-giang replied:
Fixed! Thanks for picking it up!

BipBopBoop commented on portfolio/src/main/webapp/script.js:
```diff
@@ -13,16 +13,41 @@
+
+/**
+ * Show the full blog post text for blog post div with the given ID.
+ * @param {*} blogId ID of the div of the blog post to be shown.
```
nit: We should include the param type within the `{}`

anna-giang replied:
Done! I was wondering what the { } was for! 😅

BipBopBoop replied:
Thanks! You can look it, and other style guide suggestions, here: https://google.github.io/styleguide/jsguide.html#jsdoc-method-and-function-comments

BipBopBoop reviewed (changes requested):
Looks good so far. Just some nits (nits are more just nitpicking minor details, that could potentially just be ignored).

anna-giang committed:
Fix formatting and fix minor content mistakes

- 2-space indents instead of tabs across all files
- Fix comments in script.js - add param type between {} in function header
- Blank line at the end of every file
- Fix mispelling "Hear more about *my* the..." in index.html
- Close <div id="content"> in gallery.html
- Modify <ul> in aboutMe.html - nest  all <span> under <li>

anna-giang committed:
Fix formatting of script.js according to style guide

anna-giang committed:
Replace double quotes with single quotes in script.js for string literals

BipBopBoop commented on portfolio/src/main/webapp/gallery.html:
```diff
@@ -0,0 +1,47 @@
+      </div>
+    </div>
+  </body>
+</html>
```
nit: github is suggesting that this file doesn't have a newline. It's a bit hard for me to see the line endings of files via github, so just double checking.

anna-giang replied:
It should have a blank line now

BipBopBoop replied:
Yep, seems to be there now. Thanks :)

BipBopBoop reviewed (commented):
Thanks for making the changes, looks good.

BipBopBoop reviewed (approved):
Pre-approving this % the changes requested in the last comment. (Just double check the line-endings and then everything seems good to me :) )

P.S
Sometimes a reviewer will say "LGTM % nits", this generally means that they're saying that once the nits/changes they requested are fixed they would be happy with the code. The expectation is that the owner will fix up the requested changes before merging the PR to the main branch.

anna-giang committed:
Add blank line at end of gallery.html
